content/posts/2022-01-29-things-i-learned-in-2021.md

```diff
@@ -0,0 +1,116 @@
+---
+template: post
+title: Things I learned in 2021
+slug: learn-2021
+draft: false
+date: 2022-01-29T18:13:34.496Z
+description: About what I learned in 2021.
+category: Blog
+tags:
+  - open source
+  - typescript
+  - css
+  - rust
+  - wireshark
+  - nextjs
+  - ctf
+  - web security
+  - aws
+---
+
+At the beginning of 2021, I knew that I’m in the middle of Valley of Despair of the [Dunning-Kruger Effect](https://en.wikipedia.org/wiki/Dunning%E2%80%93Kruger_effect). I also realized that I was having a lot of gaps in my knowledge. So I forced myself to learn and practice harder than previous years to fill these holes. In this post, I want to share something that I did and learned through the year, with some context on each item. I think it’s good for me to look back, and it also helps me to plan better this year (2022). And the next reason is that I love sharing 😬
+
```

Comment:
You should provide the table of contents here. For example,

To give you a better overview, these are the topics that I would share in the upcoming parts!
- Update my open source
- ....

Reply:
I also suggest dividing topics based on some specific units such as Learning new things, Open Source exploration, Improve my work experience, etc.
OR
List out based on time!
I think it is also helpful to provide how much time you spend on each task (You can draw a diagram to show a year's improvement).